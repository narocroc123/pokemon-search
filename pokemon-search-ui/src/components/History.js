import styled from "styled-components";
import fetchData from "../utils/api";
import { updateSearch } from "../actions/index";
import { useSelector } from "react-redux";
import { useDispatch } from "react-redux";

const Button = styled.input`
  background: #0066a2;
  color: white;
  border-color: #0066a2;
  height: 40px;
  width: 100px;
  margin: 5px;
  float: left;
  font-size: inherit;
  &:hover {
    cursor: pointer;
  }
`;

const History = () => {
  const dispatch = useDispatch();
  const historyResults = useSelector((state) => state.history);

  const formattedResults =
    historyResults.length >= 5
      ? historyResults.slice(historyResults.length - 5, historyResults.length)
      : historyResults;

  return (
    <>
      <h1>Recent Searches</h1>
      {formattedResults.map((item) => (
        <ul key={item}>
          <Button
            type="button"
            onClick={async () => {
              const searchResult = await fetchData(item);
              dispatch(updateSearch(searchResult));
            }}
            value={item}
          />
        </ul>
      ))}
    </>
  );
};

export default History;
