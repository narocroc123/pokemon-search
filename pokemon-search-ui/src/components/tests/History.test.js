import React from "react";
import { render } from "@testing-library/react";

import History from "../Search";
import { Provider } from "react-redux";
import { createStore } from "redux";
import rootReducer from "../../reducers/index";

const preLoadedState = {
  search: {
    data: {
      name: "charizard",
      description:
        "Spits fire that is hot enough to melt boulders. Known to cause forest fires unintentionally.",
      is_legendary: false,
    },
  },
  history: ["pikachu", "azelf", "charizard"],
};

const mockStore = createStore(rootReducer, preLoadedState);

describe("App", () => {
  const component = (
    <Provider store={mockStore}>
      <History />
    </Provider>
  );

  test("renders App component", () => {
    render(component);
  });

  test("render to match snapshot", () => {
    const { asFragment } = render(component);

    expect(asFragment(component)).toMatchSnapshot();
  });
});
