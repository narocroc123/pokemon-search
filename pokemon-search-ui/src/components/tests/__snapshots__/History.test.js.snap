// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render to match snapshot 1`] = `
<DocumentFragment>
  <form
    class="sc-bczRLJ iNhTgm"
  >
    <input
      class="sc-gsnTZi vgHnQ"
      id="search"
      name="search"
      placeholder="Search Pokemon....."
      required=""
      type="text"
      value=""
    />
    <button
      class="sc-dkzDqf bdGbai"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
